### DISCUSSION TITLE
Extensions Ranking and Search Results

### DISCUSSION THREAD

**Discussion by RandomFractals** [microsoft/vscode-discussions#99] [Extension Development QnA] [answered]
I would like to learn more about how this new Extension Search page is being used by the vs marketplace team: 

https://extensions.ms/

The original discovery of that custom extensions search page comes from the popular vs marketplace page improvements checklist [ticket](https://github.com/microsoft/vsmarketplace/issues/154#issuecomment-1212939453).

 As an extension developer, I have some questions and concerns about this under the radar tool open to the public without any user authentication and validation:

1. Do you use it as part of some ML pipeline to improve search results in marketplace?
2. Should not you have some user authentication and validation when users of that page rearrange extensions to improve your search results?
3. Should not you add keywords, categories, and a link to the extension hover card to display full extension info summary and a link to the extension in vs marketplace?
4. What guards do you have in place for that page to prevent someone scripting and upvoting their extensions? I've seen that happen with Svg Preview before as an example by unknown entities.
5. Any other extension developers concerned that such unguarded tools and other changes being made in marketplace will adversely affect their extension search results?

As it has been for a long time, vs marketplace search engine is a black box and we don't have any insight or clarity of its inner workings. 

I think it would be valuable to the vscode extensions dev community to have a public list of rules being used in extension searches and know how extension name, description, keywords, commands and other ext. config attributes are used and weighted by the vs marketplace extension search engine, including ranking of 3rd party extensions versus those created by Microsoft.

**Comment by isidorn** [comment, marked as answer]
Please ask this on the https://github.com/microsoft/vsmarketplace since this question is for the marketplace team.
Thank you

**Reply by RandomFractals** [reply]
I did. See referenced ticket link in my original question post above. Thanks for your follow up here!
